harvard-leeds-beckett-university.csl

```diff
@@ -0,0 +1,644 @@
+<?xml version="1.0" encoding="utf-8"?>
+<style xmlns="http://purl.org/net/xbiblio/csl" version="1.0" class="in-text" default-locale="en-GB" page-range-format="expanded">
+
+<info>
+  <title>Harvard - Leeds Beckett University</title>
+  <id>http://www.zotero.org/styles/harvard-leeds-beckett-university</id>
+  <link rel="self" href="http://www.zotero.org/styles/harvard-leeds-beckett-university" />
+  <link rel="documentation" href="http://skillsforlearning.leedsbeckett.ac.uk/local/Quote_Unquote.pdf" />
+  <author>
+    <name>Joe Joiner</name>
+    <email>joe@joejoiner.me</email>
+    <uri>http://twitter.com/joejoinerr</uri>
+  </author>
+  <category citation-format="author-date" />
+  <category field="generic-base" />
+  <updated>2015-03-22T18:30:00+00:00</updated>
+  <rights license="http://creativecommons.org/licenses/by-sa/3.0/">This work is licensed under a Creative Commons Attribution-ShareAlike 3.0 License</rights>
+</info>
+
+
+<macro name="online">
+  <choose>
+    <if variable="URL">
+      <group prefix=" [" suffix="]">
+        <text term="online" text-case="capitalize-first" />
+        <choose>
+          <if type="personal_communication">
+            <text value=" discussion list" />
+          </if>
+          <else-if type="graphic">
+            <text value=" image" />
+          </else-if>
+          <else-if type="post-weblog">
+            <text value=" blog" />
+          </else-if>
+          <else-if type="motion_picture">
+            <text value=" video" />
+          </else-if>
+        </choose>
+      </group>
+    </if>
+  </choose>
+</macro>
+
+<macro name="access">
+  <choose>
+    <if variable="URL">
+      <text value="Available from: " />
+      <text variable="URL" prefix="&lt;" suffix="&gt;" />
+      <group prefix=" [" suffix="]">
+        <text term="accessed" text-case="capitalize-first" suffix=" "/>
+        <date variable="accessed" delimiter=" ">
+          <date-part name="day" />
+          <date-part name="month" />
+          <date-part name="year" />
+        </date>
+      </group>
+    </if>
+  </choose>
+</macro>
+
+<macro name="container-title">
+  <choose>
+    <if type="chapter  post-weblog  entry-dictionary  entry-encyclopedia  article-journal  article-newspaper  article-magazine  paper-conference">
+      <text variable="container-title" font-style="italic" />
+    </if>
+    <else>
+      <text variable="container-title" />
+    </else>
+  </choose>
+</macro>
+
+<macro name="edition">
+  <choose>
+    <if is-numeric="edition">
+      <number variable="edition" form="ordinal" suffix=" " />
+      <label variable="edition" form="short" />
+    </if>
+  </choose>
+</macro>
+
+<macro name="locator-range">
+  <choose>
+    <if variable="page">
```

Comment:
I think you're misunderstanding the relationship between ```page``` and ```locator``` here: this would always print the full page range for a journal article or a book chapter for the in text citation. That's almost certainly not right. Page can also just be another locator. If you just use the bit from lines 89-90, that will give you p. 80 or pp. 81-84 for specific page ranges cited.